python/paddle/tensor/math.py

```diff
@@ -5096,6 +5096,105 @@ def digamma_(x, name=None):
         return _C_ops.digamma_(x)
 
 
+def gammaincc(x, a, name=None):
+    r"""
+    Computes the regularized upper incomplete gamma function.
+
+    .. math:: Q(a, x) = \frac{1}{\Gamma(a)} \int_{x}^{\infty} t^{a-1} e^{-t} dt
+
+    Args:
+        x (Tensor): The positive parameter Tensor. Must be one of the following types: float32, float64.
+        a (Tensor): The non-negative argument Tensor. Must be one of the following types: float32, float64.
+        name (str, optional): Name for the operation (optional, default is None). For more information, please refer to :ref:`api_guide_Name`.
+
+    Returns:
+        Tensor, the gammaincc of the input Tensor.
+
+    Examples:
+        .. code-block:: python
+
+            >>> import paddle
+
+            >>> x = paddle.to_tensor([0, 1, 10, 100, 1000], dtype="float32")
+            >>> a = paddle.to_tensor([0.5, 0.5, 0.5, 0.5, 0.5], dtype="float32")
+            >>> out = paddle.gammaincc(x, a)
+            >>> print(out)
+            Tensor(shape=[5], dtype=float32, place=Place(cpu), stop_gradient=True,
+                [1.        , 0.15729916, 0.00000774, 0.        , 0.        ])
+    """
+    if not paddle.all(paddle.greater_equal(a, paddle.zeros_like(a))):
+        raise ValueError(
+            "The input argument a must be greater than or equal to 0."
+        )
+    if not paddle.all(paddle.greater_equal(x, paddle.zeros_like(x))):
+        raise ValueError(
+            "The input argument x must be greater than or equal to 0."
+        )
+    if in_dynamic_or_pir_mode():
+        return _C_ops.igamma(x, a)
+    else:
+        check_variable_and_dtype(x, 'x', ['float32', 'float64'], 'gammaincc')
+        check_variable_and_dtype(a, 'a', ['float32', 'float64'], 'gammaincc')
+        helper = LayerHelper('gammaincc', **locals())
+        out = helper.create_variable_for_type_inference(x.dtype)
+        helper.append_op(
+            type='igamma', inputs={'x': x, 'a': a}, outputs={'out': out}
+        )
+        return out
+
+
+@inplace_apis_in_dygraph_only
+def gammaincc_(x, a, name=None):
+    r"""
+    Inplace version of ``gammaincc`` API, the output Tensor will be inplaced with input ``x``.
+    Please refer to :ref:`api_paddle_gammaincc`.
+    """
+    if in_dynamic_mode():
+        return _C_ops.igamma_(x, a)
+
+
+def gammainc(x, a, name=None):
```

Comment:
It is best to have the API parameter order consistent with the formula `P(a, x)` parameter order for easy understanding. according to the [API design and naming conventions](https://www.paddlepaddle.org.cn/documentation/docs/zh/dev_guides/api_contributing_guides/api_design_guidelines_standard_cn.html), suggest API design as `gammainc(x, y, name=None)`, and formula as P(x, y) = \frac{1}{\Gamma(x)} \int_{0}^{y} t^{x-1} e^{-t} dt

Reply:
done